### PR TITLE
refactor: remove dead iterator code

### PR DESCRIPTION
**Pull request by lesam**
### Required checklist
- [ ] Sample config files updated (both `/etc` folder and `NewDemoConfig` methods) (influxdb and plutonium)
- [ ] openapi swagger.yml updated (if modified API) - link openapi PR
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### Description
This removed iterator code was not used except in tests. I changed the tests to test the iterator we actually use.

### Context
N/A

### Affected areas (delete section if not relevant):
N/A

### Severity (delete section if not relevant)
N/A

### Note for reviewers:
Check the semantic commit type:
 - Feat: a feature with user-visible changes
 - Fix: a bug fix that we might tell a user “upgrade to get this fix for your issue”
 - Chore: version bumps, internal doc (e.g. README) changes, code comment updates, code formatting fixes… must not be user facing (except dependency version changes)
 - Build: build script changes, CI config changes, build tool updates
 - Refactor: non-user-visible refactoring
 - Check the PR title: we should be able to put this as a one-liner in the release notes

### CODE REVIEW

**Review by davidby-influx — APPROVED** [review]
LGTM

File: tsdb/engine/tsm1/engine.go [hunk 1]
```diff
@@ -44,8 +44,8 @@ import (
 // to support adding templated data from the command line.
 // This can probably be worked into the upstream tmpl
 // but isn't at the moment.
-//go:generate go run ../../../tools/tmpl -i -data=file_store.gen.go.tmpldata file_store.gen.go.tmpl=file_store.gen.go
```
Comment (jeffreyssmith2nd) [inline tsdb/engine/tsm1/engine.go:47]: Why did we need the change here and in `file_store.gen.go.tmpl`?

Comment (lesam) [reply tsdb/engine/tsm1/engine.go:47]: `-i` says to run `goimports` on the generated code.

We don't actually need `goimports` (the change inside `file_store.gen.go.tmpl` below makes us have the right imports) and the `goimports` is just whatever binary happens to be on your path, which is annoying when running the codegen for the first time if you don't happen to have `goimports` on your PATH at all.

Comment (jeffreyssmith2nd) [reply tsdb/engine/tsm1/engine.go:47]: Makes sense thanks 👍

**Review by jeffreyssmith2nd — APPROVED** [review]

**Comment by lesam** [comment]
Re-opening pull request to kick circleci jobs - I tried triggering a re-run in circle but that messed something up

**Comment by lesam** [comment]
And of course it's because I've made the PR from a fork. I've pushed the branch to upstream to kick of final CI.

**Pull request merged**